source/module_io/write_cube.cpp

```diff
@@ -127,8 +127,43 @@ void ModuleIO::write_cube(
         ofs_cube << std::scientific;
     }
 
+#ifdef __MPI
+    ModuleIO::write_cube_core(ofs_cube, bz, nbz, nplane, startz_current, data, nx*ny, nz, 6);
+#else
+    ModuleIO::write_cube_core(ofs_cube, data, nx*ny, nz, 6);
+#endif
+
+    if (my_rank == 0)
+    {
+        end = time(NULL);
+        ModuleBase::GlobalFunc::OUT_TIME("write_cube", start, end);
+
+        /// for cube file
+        ofs_cube.close();
+    }
+
+    return;
+}
+
+
+void ModuleIO::write_cube_core(
```

Comment:
From your PR I can realize that maybe the original implementation of writing or even together with reading the cube file really is/are bad, it is the reason why you have to implement different functions to write cube file, even if the way to organize the cube file is almost unique...what should do is to reorganize the data to fit the structure that required by the cube file before calling a function to really write the cube file, I think.